[Streams] Add classic stream creation flyout to Streams page

damian-polewski opened the pull request:
## Summary

Closes #241646

This PR adds classic stream creation UI to the Streams.

### Changes

- Adds `Create classic stream` button to Streams page that opens `CreateClassicStreamFlyout`,
- Adds `POST /internal/streams/_create_classic` endpoint to handle classic stream creation,
- Adds `POST /internal/streams/_validate_classic_stream` endpoint to handle duplicate name and higher priority matching template validation,
- Adds helper functions (`indexPatternToRegex`, `matchesPattern`, `findConflictingTemplates`) to find conflicting templates,
- Changes feedback button on the main Streams page to icon button.
- Moves `INDEX_MANAGEMENT_LOCATOR_ID` to shared index management package to fix circular dependencies,
- Copies `TemplateListItem` type to shared index management package and used it in streams_app and classic_stream_flyout since this is the type returned by `GET /api/index_management/index_templates` which we are using,
- Extends `indexManagement.apiService` to allow fetching all Index Templates by using `getIndexTemplates()`,
- Adds `showDataRetention` prop to `CreateClassicStreamFlyout`.
- Adds `isSubmitting` to `formReducer` state.

### How to test

#### Standard flow

1. Navigate to Streams
2. Click `Create classic stream` button
3. Select the template
4. Name the stream and confirm

#### Higher priority validation errors

1. To test higher priority validation errors create two templates with different priorities

```
PUT /_index_template/lower_priority
{
  "index_patterns" : ["*-logs-*-*"],
  "priority" : 1,
  "data_stream": {}
}

PUT /_index_template/higher_priority
{
  "index_patterns" : ["foo-logs-bar-*"],
  "priority" : 2,
  "data_stream": {}
}
```

2. Navigate to Streams
3. Click `Create classic stream` button
4. Select the `lower_priority` template
5. Enter `foo-logs-bar-baz` stream name
6. Try to create the classic stream

### Demo

https://github.com/user-attachments/assets/add6e87a-7e98-4faa-9ac7-bbc6885e51b1


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

## Release note

Adds a new flyout to create classic streams directly from the Streams page. The UI includes validation to prevent duplicate stream names and detect index template priority conflicts and ensuring streams are created with the expected configuration.

damian-polewski committed:
wip

damian-polewski committed:
feat: add validation endpoint

damian-polewski committed:
fix: simplify creation params

kibanamachine committed:
Changes from node scripts/lint_ts_projects --fix

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/372057)
* Commit: 58b29e5132bd2215a1267ac89375d1dd386101d6

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/372057#019b0ec6-79ee-4425-a18e-8f6f6eb8e3ab)

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372057","number":372057,"commit":"58b29e5132bd2215a1267ac89375d1dd386101d6"}],"number":372057}
buildkite-pr-comment-->

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/372068)
* Commit: 10089ea9454a586357c93e241ce67510238b6f39

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/372068#019b0edf-20d2-43e8-a87b-88f0ca4fce0b)

### History
* :broken_heart: [Build #372057](https://buildkite.com/elastic/kibana-pull-request/builds/372057) failed 58b29e5132bd2215a1267ac89375d1dd386101d6

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372068","number":372068,"commit":"10089ea9454a586357c93e241ce67510238b6f39"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372057","number":372057,"commit":"58b29e5132bd2215a1267ac89375d1dd386101d6"}],"number":372068}
buildkite-pr-comment-->

damian-polewski committed:
Merge branch 'main' into feat/add_classic_stream_creation_flyout_to_streams

flash1293 commented on x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/index.tsx:
```diff
@@ -145,7 +155,7 @@ export function StreamListView() {
               </EuiButtonEmpty>
             </EuiFlexItem>
-            <FeedbackButton />
+            <FeedbackButton iconButton />
```
I remember something about this feedback button and that it should always be rendered in the same way. @patpascal do you know what I mean?

damian-polewski replied:
In the designs there were only changes for the button on the main page, but if we need I can update it everywhere. @patpascal @MichaelMarcialis wdyt?

flash1293 commented on x-pack/platform/plugins/shared/streams/server/routes/streams/crud/validate_classic_stream_route.ts:
```diff
@@ -0,0 +1,110 @@
+
+    // Check 2: Does a data stream with this name already exist?
+    try {
+      const dataStreamResult = await scopedClusterClient.asCurrentUser.indices.getDataStream({
```
This shouldn't be necessary - `streamsClient.existsStream` should handle this case already

flash1293 reviewed (commented):
The general setup makes sense for me for now - eventually I think we should have a public API to create a classic stream and not just an internal one, but it can be a follow-up, so no need to tackle it now.

I left some smaller nits, but overall good work.

Doesn't need to happen on this PR, but we should add a basic smoke test via scout to make sure this doesn't break.

damian-polewski committed:
fix: add simplified index template type to avoid circular dependencies

damian-polewski committed:
fix: use locator instead of hardcoded path

damian-polewski committed:
fix: remove unnecessary data stream cleanup

SoniaSanzV reviewed (dismissed):
Flyout changes LGTM!

damian-polewski committed:
fix: remove unnecessary data stream check

damian-polewski committed:
feat: expose api clients and use in streams

kibanamachine committed:
Changes from node scripts/lint_ts_projects --fix

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/372763)
* Commit: f5d925d89d0ee75ff051b9c26aefe5956eb1bae4

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/372763#019b14cd-ef52-4d44-afae-024ba29d98e3)

### History
* :broken_heart: [Build #372756](https://buildkite.com/elastic/kibana-pull-request/builds/372756) failed 7c0ba966a185cc32260fe1d89d18c03ef935daba
* :broken_heart: [Build #372508](https://buildkite.com/elastic/kibana-pull-request/builds/372508) failed 4b555682edec316f5bad63ca8dcb7add4f1c1bc4
* :broken_heart: [Build #372470](https://buildkite.com/elastic/kibana-pull-request/builds/372470) failed 09175ec99a7f182f1cebc41968564807dd6b34ca
* :broken_heart: [Build #372394](https://buildkite.com/elastic/kibana-pull-request/builds/372394) failed e58b982ed38f9153766d39dfb3efe54e5eb4c68b
* :broken_heart: [Build #372236](https://buildkite.com/elastic/kibana-pull-request/builds/372236) failed 17c9d3b734c4fdaa65ae043553ed594bda93df24

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372763","number":372763,"commit":"f5d925d89d0ee75ff051b9c26aefe5956eb1bae4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372756","number":372756,"commit":"7c0ba966a185cc32260fe1d89d18c03ef935daba"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372508","number":372508,"commit":"4b555682edec316f5bad63ca8dcb7add4f1c1bc4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372470","number":372470,"commit":"09175ec99a7f182f1cebc41968564807dd6b34ca"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372394","number":372394,"commit":"e58b982ed38f9153766d39dfb3efe54e5eb4c68b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372236","number":372236,"commit":"17c9d3b734c4fdaa65ae043553ed594bda93df24"}],"number":372763}
buildkite-pr-comment-->

damian-polewski committed:
Merge branch 'main' into feat/add_classic_stream_creation_flyout_to_streams

damian-polewski committed:
fix: move locator to shared package to fix circular dependency

damian-polewski committed:
fix: copy template list item to shared and unify used types in streams and classic stream flyout package

damian-polewski committed:
Merge branch 'main' into feat/add_classic_stream_creation_flyout_to_streams

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

damian-polewski commented:
Hi @flash1293 @SoniaSanzV, thank you for reviews!

@flash1293 I addressed the feedback, could you please take a look? I will create a follow up PR adding Scout tests for this flow to unblock this PR if that's okay.

@SoniaSanzV I had to add some additional changes to kibana-management's packages to fix the circular dependencies issues, could you please review them and let me know what do you think? 
Changes summary:

- Moved `INDEX_MANAGEMENT_LOCATOR_ID` to shared index management package to fix circular dependencies,
- Copied `TemplateListItem` type to shared index management package and used it in streams_app and classic_stream_flyout since this is the type returned by `GET /api/index_management/index_templates` which we are using,
- Added ILM api client (`createIlmApiClient`) to expose API for fetching ILM policies,
- Extended `indexManagement.apiService` to allow fetching all Index Templates by using `getIndexTemplates()`.

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/372854)
* Commit: 486fa22fd2153779b84ffb3123b77a420b815e37
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-245975)

### Failed CI Steps
* [FTR Configs #59](https://buildkite.com/elastic/kibana-pull-request/builds/372854#019b1f26-894a-4d30-b948-a195e9210951)
* [FTR Configs #70](https://buildkite.com/elastic/kibana-pull-request/builds/372854#019b1f26-8957-4a5e-a981-6390a7bf4851)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/372854#019b1f26-894a-4d30-b948-a195e9210951) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/372854/jobs/019b1f26-894a-4d30-b948-a195e9210951/artifacts/019b1f3f-193e-4762-ac04-41d38de0084f) FTR Configs #<span></span>59 / integrations When on the Endpoint Policy List Page with no policies shows the empty page
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/372854#019b1f26-8957-4a5e-a981-6390a7bf4851) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/372854/jobs/019b1f26-8957-4a5e-a981-6390a7bf4851/artifacts/019b1f50-5361-4389-b428-b43c21cac31b) FTR Configs #<span></span>70 / serverless observability UI - Cases and Rules Serverless Observability Cases Cases list row actions Severity to medium

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/6db92952dae43cff356ecf0fa19e85f9bd035cbf) | [after](https://github.com/elastic/kibana/commit/486fa22fd2153779b84ffb3123b77a420b815e37) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 187 | 188 | +1 |
| `indexManagement` | 683 | 685 | +2 |
| `streamsApp` | 1304 | 1335 | +31 |
| `synthetics` | 1372 | 1373 | +1 |
| total |  |  | +35 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/6db92952dae43cff356ecf0fa19e85f9bd035cbf) | [after](https://github.com/elastic/kibana/commit/486fa22fd2153779b84ffb3123b77a420b815e37) | diff |
| --- | --- | --- | --- |
| `@kbn/index-lifecycle-management-common-shared` | 76 | 82 | +6 |
| `@kbn/index-management-shared-types` | 148 | 180 | +32 |
| total |  |  | +38 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6db92952dae43cff356ecf0fa19e85f9bd035cbf) | [after](https://github.com/elastic/kibana/commit/486fa22fd2153779b84ffb3123b77a420b815e37) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 1.4MB | 1.4MB | +27.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/6db92952dae43cff356ecf0fa19e85f9bd035cbf) | [after](https://github.com/elastic/kibana/commit/486fa22fd2153779b84ffb3123b77a420b815e37) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 32.1KB | 32.5KB | +392.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/6db92952dae43cff356ecf0fa19e85f9bd035cbf) | [after](https://github.com/elastic/kibana/commit/486fa22fd2153779b84ffb3123b77a420b815e37) | diff |
| --- | --- | --- | --- |
| `@kbn/index-lifecycle-management-common-shared` | 78 | 85 | +7 |
| `@kbn/index-management-shared-types` | 148 | 183 | +35 |
| `indexManagement` | 260 | 261 | +1 |
| total |  |  | +43 |


</details>


### History
* :broken_heart: [Build #372847](https://buildkite.com/elastic/kibana-pull-request/builds/372847) failed 357238fde61c8fd394c9ad0368da7ba716f642f5
* :broken_heart: [Build #372778](https://buildkite.com/elastic/kibana-pull-request/builds/372778) failed 51eb4b15ab2c9d4701227e7be5fc8de6a4de8248
* :broken_heart: [Build #372763](https://buildkite.com/elastic/kibana-pull-request/builds/372763) failed f5d925d89d0ee75ff051b9c26aefe5956eb1bae4
* :broken_heart: [Build #372756](https://buildkite.com/elastic/kibana-pull-request/builds/372756) failed 7c0ba966a185cc32260fe1d89d18c03ef935daba
* :broken_heart: [Build #372508](https://buildkite.com/elastic/kibana-pull-request/builds/372508) failed 4b555682edec316f5bad63ca8dcb7add4f1c1bc4
* :broken_heart: [Build #372470](https://buildkite.com/elastic/kibana-pull-request/builds/372470) failed 09175ec99a7f182f1cebc41968564807dd6b34ca

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372854","number":372854,"commit":"486fa22fd2153779b84ffb3123b77a420b815e37"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372847","number":372847,"commit":"357238fde61c8fd394c9ad0368da7ba716f642f5"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372778","number":372778,"commit":"51eb4b15ab2c9d4701227e7be5fc8de6a4de8248"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372763","number":372763,"commit":"f5d925d89d0ee75ff051b9c26aefe5956eb1bae4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372756","number":372756,"commit":"7c0ba966a185cc32260fe1d89d18c03ef935daba"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372508","number":372508,"commit":"4b555682edec316f5bad63ca8dcb7add4f1c1bc4"}],"number":372854}
buildkite-pr-comment-->

damian-polewski committed:
fix: error handling in handleValidate

damian-polewski committed:
fix: feedback button changes

SoniaSanzV commented on x-pack/platform/plugins/shared/index_management/public/services/public_api_service.ts:
```diff
@@ -32,4 +33,13 @@ export class PublicApiService {
+  getIndexTemplates(options?: { signal?: AbortSignal }) {
+    return this.http.get<GetIndexTemplatesResponse>(`${API_BASE_PATH}/index_templates`, {
+      signal: options?.signal,
+    });
```
`sendRequest` is ours. Can you update it to accept an optional signal prop so it can be used here?

  getIndexTemplates(options?: { signal?: AbortSignal }) {
    return sendRequest<GetIndexTemplatesResponse>(this.http, {
      path: `${API_BASE_PATH}/index_templates`,
      method: 'get',
      signal: options?.signal,
    });
  }

damian-polewski committed:
fix: invalid state missing right outline

patpascal reviewed (approved):
LGTM

flash1293 reviewed (commented):
Overall looks pretty good, it would be great to have a super basic smoke test for the functionality to make sure it doesn't break (just the for the happy path)

Loading states are not handled very well, check this recording with delaying all the requests. I think we can improve here a bit:

![Image](https://github.com/user-attachments/assets/a52a70af-f99a-470b-9bbd-3c0dd05a37fd)

Probably the most important part - I'm not sure we should show the retention here like this:

<img width="633" height="481" alt="Image" src="https://github.com/user-attachments/assets/24bad7eb-af82-4e30-8a25-1610a6cdde37" />

This is a bit misleading - these index templates could totally have some retention set, but it's hidden in a component template, so it doesn't show in this list.


Last one is a pure UX nit - it's kind of weird I have to click next after selecting a single item from the list - why are you forcing me to click again? It should just move to the next step immediately. But that's not so important.

damian-polewski committed:
fix: add spinner when loading templates

damian-polewski committed:
fix: navigate to first tab after creation

damian-polewski committed:
feat: add scout test for classic stream creation flow

damian-polewski committed:
feat: expose ilm api service instead of exporting api client

damian-polewski commented on x-pack/platform/packages/shared/index-lifecycle-management/index_lifecycle_management_common_shared/src/api_client.ts:
```diff
@@ -0,0 +1,28 @@
+/**
+ * Creates a minimal typed ILM API client for cross-plugin usage.
+ */
+export function createIlmApiClient(core: CoreStart | CoreSetup): IlmApiClient {
```
Addressed in https://github.com/elastic/kibana/pull/245975/commits/978b715e645012d5cd2960cf1da1cfdf8cbedf1a

damian-polewski committed:
Merge branch 'main' into feat/add_classic_stream_creation_flyout_to_streams

kibanamachine committed:
Changes from node scripts/lint_ts_projects --fix

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/373375)
* Commit: 69379f10b93b6418ec3c8a70bdf8c5f13d53f71f

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/373375#019b230a-3090-496c-9e81-9980de1e7aeb)

### History
* :broken_heart: [Build #373349](https://buildkite.com/elastic/kibana-pull-request/builds/373349) failed 44d4090e8e1b5c3d6f6856c066c3777e4114ae53
* :broken_heart: [Build #373111](https://buildkite.com/elastic/kibana-pull-request/builds/373111) failed dd5b182ffabab13e02f01e66d0b7729e0e3e4d91
* :broken_heart: [Build #373019](https://buildkite.com/elastic/kibana-pull-request/builds/373019) failed 7fbec16197b9ba41556fe58aa1d45fabec6f22a4
* :yellow_heart: [Build #372854](https://buildkite.com/elastic/kibana-pull-request/builds/372854) was flaky 486fa22fd2153779b84ffb3123b77a420b815e37
* :broken_heart: [Build #372847](https://buildkite.com/elastic/kibana-pull-request/builds/372847) failed 357238fde61c8fd394c9ad0368da7ba716f642f5

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373375","number":373375,"commit":"69379f10b93b6418ec3c8a70bdf8c5f13d53f71f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373349","number":373349,"commit":"44d4090e8e1b5c3d6f6856c066c3777e4114ae53"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373111","number":373111,"commit":"dd5b182ffabab13e02f01e66d0b7729e0e3e4d91"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373019","number":373019,"commit":"7fbec16197b9ba41556fe58aa1d45fabec6f22a4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372854","number":372854,"commit":"486fa22fd2153779b84ffb3123b77a420b815e37"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372847","number":372847,"commit":"357238fde61c8fd394c9ad0368da7ba716f642f5"}],"number":373375}
buildkite-pr-comment-->

damian-polewski committed:
fix: import circular dependency

kibanamachine committed:
Changes from node scripts/lint_ts_projects --fix

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/373500)
* Commit: e85823f35ace5b4044ca667c663eee49e6e582d3

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/373500#019b23c9-1a7b-4118-9630-f0ecb45d6952)

### History
* :broken_heart: [Build #373486](https://buildkite.com/elastic/kibana-pull-request/builds/373486) failed 33d7d89fc11216e430b5f61dc576364eebe51d07
* :broken_heart: [Build #373375](https://buildkite.com/elastic/kibana-pull-request/builds/373375) failed 69379f10b93b6418ec3c8a70bdf8c5f13d53f71f
* :broken_heart: [Build #373349](https://buildkite.com/elastic/kibana-pull-request/builds/373349) failed 44d4090e8e1b5c3d6f6856c066c3777e4114ae53
* :broken_heart: [Build #373111](https://buildkite.com/elastic/kibana-pull-request/builds/373111) failed dd5b182ffabab13e02f01e66d0b7729e0e3e4d91
* :broken_heart: [Build #373019](https://buildkite.com/elastic/kibana-pull-request/builds/373019) failed 7fbec16197b9ba41556fe58aa1d45fabec6f22a4

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373500","number":373500,"commit":"e85823f35ace5b4044ca667c663eee49e6e582d3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373486","number":373486,"commit":"33d7d89fc11216e430b5f61dc576364eebe51d07"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373375","number":373375,"commit":"69379f10b93b6418ec3c8a70bdf8c5f13d53f71f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373349","number":373349,"commit":"44d4090e8e1b5c3d6f6856c066c3777e4114ae53"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373111","number":373111,"commit":"dd5b182ffabab13e02f01e66d0b7729e0e3e4d91"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373019","number":373019,"commit":"7fbec16197b9ba41556fe58aa1d45fabec6f22a4"}],"number":373500}
buildkite-pr-comment-->

damian-polewski committed:
Merge branch 'main' into feat/add_classic_stream_creation_flyout_to_streams

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/373524)
* Commit: d226eec21884d3ec0f576b7424c2622808eb73ee
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-245975)

### Failed CI Steps
* [FTR Configs #66](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b242e-c5fd-4e18-9469-7c3118ae277e)
* [FTR Configs #66](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b2456-7646-41f1-874a-118d065d1bd8)
* [FTR Configs #105](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b242e-c624-48a5-8adc-559289ddc059)
* [FTR Configs #105](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b2459-9e98-474c-8864-e1c4fc972f03)
* [FTR Configs #106](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b242e-c625-4f29-9a1c-b2854595a88b)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b242e-c625-4f29-9a1c-b2854595a88b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373524/jobs/019b242e-c625-4f29-9a1c-b2854595a88b/artifacts/019b244f-d38f-42a2-ae8f-913b3ddfc11c) FTR Configs #<span></span>106 / Dataset Quality Dataset Quality Details failed docs should go to discover for failed docs when the button next to breakdown selector is clicked
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b242e-c625-4f29-9a1c-b2854595a88b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373524/jobs/019b242e-c625-4f29-9a1c-b2854595a88b/artifacts/019b244f-d38e-4e03-bf26-31b0dec7cfea) FTR Configs #<span></span>106 / Endpoint Exceptions with feature flag enabled @<span></span>ess @<span></span>serverless @<span></span>skipInServerlessMKI Endpoint artifacts (via lists plugin): Endpoint Exceptions "before all" hook in "@<span></span>ess @<span></span>serverless @<span></span>skipInServerlessMKI Endpoint artifacts (via lists plugin): Endpoint Exceptions"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b242e-c624-48a5-8adc-559289ddc059) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373524/jobs/019b242e-c624-48a5-8adc-559289ddc059/artifacts/019b2459-40ba-4323-a7fe-a15ec8e85e63) FTR Configs #<span></span>105 / serverless observability logs essentials UI logs essentials only navigation - streams entry shows the Streams entry in side nav (experimental)
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b2459-9e98-474c-8864-e1c4fc972f03) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373524/jobs/019b2459-9e98-474c-8864-e1c4fc972f03/artifacts/019b2462-1ec2-4a34-963a-079714a396c4) FTR Configs #<span></span>105 / serverless observability logs essentials UI logs essentials only navigation - streams entry shows the Streams entry in side nav (experimental)
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b242e-c5fd-4e18-9469-7c3118ae277e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373524/jobs/019b242e-c5fd-4e18-9469-7c3118ae277e/artifacts/019b2456-49eb-40af-ba0c-f7584ed16335) FTR Configs #<span></span>66 / serverless observability UI - feature flags Observability Infra Infra Side Navigation when Hosts settings is on "before all" hook for "shows the 'Hosts' nav item"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373524#019b2456-7646-41f1-874a-118d065d1bd8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373524/jobs/019b2456-7646-41f1-874a-118d065d1bd8/artifacts/019b245f-73ff-4c4d-89d2-3d88731f84ab) FTR Configs #<span></span>66 / serverless observability UI - feature flags Observability Infra Infra Side Navigation when Hosts settings is on "before all" hook for "shows the 'Hosts' nav item"

### History
* :broken_heart: [Build #373500](https://buildkite.com/elastic/kibana-pull-request/builds/373500) failed e85823f35ace5b4044ca667c663eee49e6e582d3
* :broken_heart: [Build #373486](https://buildkite.com/elastic/kibana-pull-request/builds/373486) failed 33d7d89fc11216e430b5f61dc576364eebe51d07
* :broken_heart: [Build #373375](https://buildkite.com/elastic/kibana-pull-request/builds/373375) failed 69379f10b93b6418ec3c8a70bdf8c5f13d53f71f
* :broken_heart: [Build #373349](https://buildkite.com/elastic/kibana-pull-request/builds/373349) failed 44d4090e8e1b5c3d6f6856c066c3777e4114ae53
* :broken_heart: [Build #373111](https://buildkite.com/elastic/kibana-pull-request/builds/373111) failed dd5b182ffabab13e02f01e66d0b7729e0e3e4d91

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373524","number":373524,"commit":"d226eec21884d3ec0f576b7424c2622808eb73ee"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373500","number":373500,"commit":"e85823f35ace5b4044ca667c663eee49e6e582d3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373486","number":373486,"commit":"33d7d89fc11216e430b5f61dc576364eebe51d07"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373375","number":373375,"commit":"69379f10b93b6418ec3c8a70bdf8c5f13d53f71f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373349","number":373349,"commit":"44d4090e8e1b5c3d6f6856c066c3777e4114ae53"}],"number":373524}
buildkite-pr-comment-->

damian-polewski committed:
feat: add showDataRetention prop to CreateClassicStreamFlyout

damian-polewski committed:
Merge branch 'main' into feat/add_classic_stream_creation_flyout_to_streams

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/373572)
* Commit: 086b92411f6a8b1acca7c787cba98213a6a4e210
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-245975)

### Failed CI Steps
* [FTR Configs #29](https://buildkite.com/elastic/kibana-pull-request/builds/373572#019b2490-5077-4682-90fc-b23b1099bfcf)
* [FTR Configs #29](https://buildkite.com/elastic/kibana-pull-request/builds/373572#019b24b3-ad3d-4fe4-b0ae-19555f3f748e)
* [FTR Configs #65](https://buildkite.com/elastic/kibana-pull-request/builds/373572#019b2490-50a0-456c-a473-c5cc88756553)
* [FTR Configs #65](https://buildkite.com/elastic/kibana-pull-request/builds/373572#019b24b1-3725-4f05-a267-5c061095ae61)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373572#019b2490-5077-4682-90fc-b23b1099bfcf) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373572/jobs/019b2490-5077-4682-90fc-b23b1099bfcf/artifacts/019b24b3-7f19-400c-bd20-ad8aa0cd917c) FTR Configs #<span></span>29 / serverless observability logs essentials UI logs essentials only navigation - streams entry shows the Streams entry in side nav (experimental)
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373572#019b24b3-ad3d-4fe4-b0ae-19555f3f748e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373572/jobs/019b24b3-ad3d-4fe4-b0ae-19555f3f748e/artifacts/019b24be-0bb8-43a9-9ce4-195c8643a29d) FTR Configs #<span></span>29 / serverless observability logs essentials UI logs essentials only navigation - streams entry shows the Streams entry in side nav (experimental)
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373572#019b2490-50a0-456c-a473-c5cc88756553) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373572/jobs/019b2490-50a0-456c-a473-c5cc88756553/artifacts/019b24b1-2008-4654-a7bc-738824720515) FTR Configs #<span></span>65 / serverless observability UI - feature flags Observability Infra Infra Side Navigation when Hosts settings is on "before all" hook for "shows the 'Hosts' nav item"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373572#019b24b1-3725-4f05-a267-5c061095ae61) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373572/jobs/019b24b1-3725-4f05-a267-5c061095ae61/artifacts/019b24b9-eaaa-492e-a7df-45fd8335ef0b) FTR Configs #<span></span>65 / serverless observability UI - feature flags Observability Infra Infra Side Navigation when Hosts settings is on "before all" hook for "shows the 'Hosts' nav item"

### History
* :broken_heart: [Build #373524](https://buildkite.com/elastic/kibana-pull-request/builds/373524) failed d226eec21884d3ec0f576b7424c2622808eb73ee
* :broken_heart: [Build #373500](https://buildkite.com/elastic/kibana-pull-request/builds/373500) failed e85823f35ace5b4044ca667c663eee49e6e582d3
* :broken_heart: [Build #373486](https://buildkite.com/elastic/kibana-pull-request/builds/373486) failed 33d7d89fc11216e430b5f61dc576364eebe51d07
* :broken_heart: [Build #373375](https://buildkite.com/elastic/kibana-pull-request/builds/373375) failed 69379f10b93b6418ec3c8a70bdf8c5f13d53f71f
* :broken_heart: [Build #373349](https://buildkite.com/elastic/kibana-pull-request/builds/373349) failed 44d4090e8e1b5c3d6f6856c066c3777e4114ae53

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373572","number":373572,"commit":"086b92411f6a8b1acca7c787cba98213a6a4e210"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373524","number":373524,"commit":"d226eec21884d3ec0f576b7424c2622808eb73ee"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373500","number":373500,"commit":"e85823f35ace5b4044ca667c663eee49e6e582d3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373486","number":373486,"commit":"33d7d89fc11216e430b5f61dc576364eebe51d07"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373375","number":373375,"commit":"69379f10b93b6418ec3c8a70bdf8c5f13d53f71f"}],"number":373572}
buildkite-pr-comment-->

damian-polewski committed:
Revert "fix: import circular dependency"

This reverts commit 33d7d89fc11216e430b5f61dc576364eebe51d07.

damian-polewski committed:
Revert "feat: expose ilm api service instead of exporting api client"

This reverts commit 978b715e645012d5cd2960cf1da1cfdf8cbedf1a.

Kerry350 reviewed (approved):
Thanks for addressing Joe's feedback. I'd suggest we defer back and forth over "next" to a followup potentially. This looks okay to me otherwise. 

**Please fix the internationalisation issues before merging, but I'm approving given the feature freeze scramble.**

damian-polewski committed:
remove ilm api client

damian-polewski committed:
fix: remove getIlmPolicy callback

damian-polewski committed:
fix: remove template name truncation when hiding data retention

damian-polewski committed:
fix: improve isSubmitting state and add createMockTemplate helper

damian-polewski committed:
fix: use i18n for toast messages

damian-polewski committed:
Merge branch 'main' into feat/add_classic_stream_creation_flyout_to_streams

SoniaSanzV reviewed (approved):
Thank you for addressing all the feedback! Tested locally, work as expected and code LGTM. Great job @damian-polewski! 🚀

MichaelMarcialis commented:
> > Last one is a pure UX nit - it's kind of weird I have to click next after selecting a single item from the list - why are you forcing me to click again? It should just move to the next step immediately. But that's not so important.
> 
> Taking the user to next step right after selecting template - I think would be a bit unintuitive considering we have a Next button. When I would see select and Next button my assumption would be that the order is: select an option -> press button to proceed. @MichaelMarcialis wdyt? If we would prefer this behavior I suggest we handle it as a follow up PR.

@SoniaSanzV: That's a fair point. Since it's a single item selection, I wouldn't be opposed to having the item selection automatically trigger moving to the next step (assuming it doesn't feel strange in practice). If we wish to make such a change, we should also remove the "Next" button from the first step. Whether we want to make this change in this PR or in a separate PR, I'll defer to you and @damian-polewski.

damian-polewski committed:
fix: ts errors
